iphone/Classes/GeolocationModule.m

```diff
@@ -879,18 +880,23 @@ -(void)requestLocationPermissions:(id)args
         if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"]) {
             if (currentPermissionLevel == kCLAuthorizationStatusAuthorizedWhenInUse) {
                 errorMessage = @"Cannot change already granted permission from AUTHORIZATION_WHEN_IN_USE to AUTHORIZATION_ALWAYS";
-            } else {
+            } else if(!userDidApproveOrReject) {
                 [[self locationPermissionManager] requestAlwaysAuthorization];
             }
         } else {
             errorMessage = @"The NSLocationAlwaysUsageDescription key must be defined in your tiapp.xml in order to request this permission.";
         }
     }
-    
-    [self executeAndReleaseCallbackWithCode:(errorMessage == nil) ? 0 : 1 andMessage:errorMessage];
-
-    if (errorMessage != nil) {
+	
+    if (userDidApproveOrReject &&
+        (currentPermissionLevel == kCLAuthorizationStatusRestricted ||
+         currentPermissionLevel == kCLAuthorizationStatusDenied)) {
+        errorMessage = @"The user denied access to use location services.";
+    }
+	
+    if (errorMessage != nil || userDidApproveOrReject) {
         NSLog(@"[ERROR] %@", errorMessage);
```

Comment:
this NSLog will appear in the case where errorMessage == nil and userDidApproveOrReject = YES (i.e. user approved).
We have to add another if statement specifically for this log if there is truly an error.